### ISSUE TITLE
postgres_x_ALL plugins produce 'FATAL: database "munin" does not exist' errors

### ISSUE THREAD

**Issue by Talkless** [munin-monitoring/munin#1108]
**Describe the bug**

This is Debian bug I've already reported:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=911656

Basically, after upgrading to `2.0.37-1~bpo9+1`, PostgreSQL clusters started to accumulate these error messages:

```
2018-10-19 16:26:20 EEST [unknown] munin munin [local] FATAL:  database "munin" does not exist
2018-10-19 16:26:20 EEST [unknown] munin munin [local] FATAL:  database "munin" does not exist
2018-10-19 16:26:20 EEST [unknown] munin munin [local] FATAL:  database "munin" does not exist
2018-10-19 16:26:21 EEST [unknown] munin munin [local] FATAL:  database "munin" does not exist
2018-10-19 16:26:21 EEST [unknown] munin munin [local] FATAL:  database "munin" does not exist
```

**Expected behavior**
Expected to work as previous 2.0.33-1 release (unless I've missed some documentation chages?)

**Screenshots & Logs**

munin-node.log contains:
```
2018/10/19-16:26:20 [3780] Error output from postgres_locks_ALL:
2018/10/19-16:26:20 [3780]      Use of uninitialized value $dbname in concatenation (.) or string at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
2018/10/19-16:26:20 [3780]      DBI connect('dbname=','',...) failed: FATAL:  database "munin" does not exist at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
2018/10/19-16:26:20 [3780] Service 'postgres_locks_ALL' exited with status 1/0.

2018/10/19-16:26:20 [3780] Error output from postgres_querylength_ALL:
2018/10/19-16:26:20 [3780]      Use of uninitialized value $dbname in concatenation (.) or string at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
2018/10/19-16:26:20 [3780]      DBI connect('dbname=','',...) failed: FATAL:  database "munin" does not exist at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
2018/10/19-16:26:20 [3780] Service 'postgres_querylength_ALL' exited with status 1/0.

2018/10/19-16:26:21 [3780] Error output from postgres_cache_ALL:
2018/10/19-16:26:21 [3780]      Use of uninitialized value $dbname in concatenation (.) or string at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
2018/10/19-16:26:21 [3780]      DBI connect('dbname=','',...) failed: FATAL:  database "munin" does not exist at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
2018/10/19-16:26:21 [3780] Service 'postgres_cache_ALL' exited with status 1/0.

2018/10/19-16:26:21 [3780] Error output from postgres_connections_ALL:
2018/10/19-16:26:21 [3780]      Use of uninitialized value $dbname in concatenation (.) or string at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
2018/10/19-16:26:21 [3780]      DBI connect('dbname=','',...) failed: FATAL:  database "munin" does not exist at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
2018/10/19-16:26:21 [3780] Service 'postgres_connections_ALL' exited with status 1/0.

2018/10/19-16:26:22 [3780] Error output from postgres_size_ALL:
2018/10/19-16:26:22 [3780]      Use of uninitialized value $dbname in concatenation (.) or string at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
2018/10/19-16:26:22 [3780]      DBI connect('dbname=','',...) failed: FATAL:  database "munin" does not exist at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
2018/10/19-16:26:22 [3780] Service 'postgres_size_ALL' exited with status 1/0.
```

My config:

```
# cat /etc/munin/plugin-conf.d/postgresql
[postgres_*]
user munin
env.PGHOST /var/run/postgresql
```

**Desktop**
 - Debian Stretch amd64
 - munin-node/plugins-core 2.0.37-1~bpo9+1

**Additional context**

Same errors on VM with only `postgres` database, and on another machine with `postgres` and some additional custom database. Same with 9.6 and 10 clusters.

**Comment by Talkless** [comment]
Looks like I have to specify default DB (I've set `template1`) to connect for `ALL` plugins:
```
[postgres_*]
user munin
env.PGDATABASE template1
env.PGHOST /var/run/postgresql
```
Is this by design..?

**Comment by steveschnepp** [comment]
The plugin tries to connect to the same default DB as `pgsql`.

So you can either:
- use the PostgreSQL `admin` user
- set any db that `munin` user can access: `template1` being a perfectly valid choice.
- create a `munin` db database.

If we provide `env.PGDATABASE template1` by default when empty, would then deviate from `pgsql`. And it would be much less obvious.

**Comment by Talkless** [comment]
> use the PostgreSQL admin user

I have set `postgres` user:
```
[postgres_*]
user postgres
env.PGHOST /var/run/postgresql
```

Then, tried to run `postgres_locks_ALL`,  and got complain `Use of uninitialized value $dbname in concatenation (.) or string at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.`:
```
# munin-run postgres_locks_ALL
Use of uninitialized value $dbname in concatenation (.) or string at /usr/share/perl5/Munin/Plugin/Pgsql.pm line 430.
accessexclusivelock.value 0
accesssharelock.value 1
exclusivelock.value 0
rowexclusivelock.value 0
rowsharelock.value 0
sharelock.value 0
sharerowexclusivelock.value 0
shareupdateexclusivelock.value 0
```

Commenting out `paramdatabase  => 1,` (as suggest in Debian bug report [0]) line in plugin fixes that warning.

[0] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=911656#12

**Comment by sumpfralle** [comment]
This issue was fixed in 3f93437b47c0d5a5cb9d450a32b8d5d5dc2cdca9 and released in 2.0.38.
